### PR TITLE
Add Online IDEs

### PR DESCRIPTION
**Pull request by styfle**
https://github.com/styfle/awesome-online-ide

A curated list of awesome online implementations of integrated development environments (IDE).

This is useful for developers who do not want to install software to start coding. For example, maybe they are trying out a new language or framework or maybe they don't have a computer that is compatible with the technologies they wish to use.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- [x] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed: PR #1369 and PR #1370
- [x] I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- [x] This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- [x] The entry in the Awesome list should:
	- [x] Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- [x] Be added at the bottom of the appropriate category.
- [x] The list I'm submitting complies with the below requirements.

## Requirements for your Awesome list

- [x] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- [x] It's the result of hard work and the best I could possibly produce.
- [x] Non-generated Markdown file in a GitHub repo.
- [x] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self) 
- [x] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [x] Not a duplicate.
- [x] Only has awesome items. Awesome lists are curations of the best, not everything.
- [x] Includes a project logo/illustration whenever possible.
	- [x] Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- [x] The image should link to the project website or any relevant website.
	- [x] The image should be high-DPI. Set it to maximum half the width of the original image.
- [x] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [x] Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- [x] Should be placed on the right side of the readme heading.
	- [x] Should link back to this list.
- [x] Has a Table of Contents section.
	- [x] Should be named `Contents`, not `Table of Contents`.
	- [x] Should be the first section in the list.
	- [x] Should only have one level of sub-lists, preferably none.
- [x] Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- [x] That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [x] [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- [x] If you use a license badge, it should be SVG, not PNG.
- [x] Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- [x] The file should be named `contributing.md`. Casing is up to you.
- [x] Has consistent formatting and proper spelling/grammar.
	- [x] The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- [x] The description starts with an uppercase character and ends with a period.
	- [x] Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [x] Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Please read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) more closely. Multiple points not followed from it. Try going through the list point for point and check that you follow it. :)

**Comment by Jogai** [comment]
see https://fiddles.io/ for more inspiration

**Comment by styfle** [comment]
@sindresorhus I added the `CC0` license, a `contributing.md` file, and a logo, and a Table of Contents. Thanks! 👍

**Comment by legodude17** [comment]
> under law, ceriously.com has

Why does it say `ceriously.com`? Just curious.

**Comment by styfle** [comment]
@legodude17 That's my organization: ceriously.com

**Comment by sindresorhus** [comment]
Hey, sorry for the lack of response, please see #1363.

**Comment by styfle** [comment]
@sindresorhus Thanks!

I went through the list several more times (even turned it into checkboxes) and everything is compliant now!

I also reviewed 2 other pull requests: PR #1369 and PR #1370

Anything else you need?

File: readme.md [hunk 1]
```diff
@@ -380,6 +380,7 @@
 - [PowerShell](https://github.com/janikvonrotz/awesome-powershell#readme) - Cross-platform object-oriented shell.
 - [Alfred Workflows](https://github.com/derimagia/awesome-alfred-workflows#readme) - Productivity app for macOS.
 - [Terminals Are Sexy](https://github.com/k4m4/terminals-are-sexy#readme)
+- [Online IDEs](https://github.com/styfle/awesome-online-ide#readme) - Online Integrated Development Environments.
```
Comment (sindresorhus) [inline readme.md]: The description should succinctly describe *what* and Online IDE is, not just what the acronym means.

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/styfle/awesome-online-ide
✖ Linting

  README.md:61:3
  ✖  15:3   Invalid list item link URL                                          remark-lint:awesome/list-item
  ✖  16:3   Invalid list item link URL                                          remark-lint:awesome/list-item
  ✖  17:1   ToC item "Web Snippets" does not match corresponding heading "RIP"  remark-lint:awesome/toc
  ✖  17:3   Invalid list item link URL                                          remark-lint:awesome/list-item
  ✖  18:3   Invalid list item link URL                                          remark-lint:awesome/list-item
  ✖  31:46  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  32:33  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  34:40  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  35:38  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  36:43  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  37:51  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  38:37  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  40:33  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  42:33  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  43:34  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  44:73  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  45:60  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  46:29  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  47:30  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  49:49  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  50:35  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  51:31  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  55:26  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  56:34  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  57:35  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  61:3   Link to https://theonlinecompiler.com is dead                       remark-lint:no-dead-urls
  ✖  65:3   Link to https://onlinegdb.com is dead                               remark-lint:no-dead-urls
  ✖  66:36  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  69:58  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  70:41  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  72:40  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  73:3   Link to https://3v4l.org is dead                                    remark-lint:no-dead-urls
  ✖  78:30  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  80:28  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  82:33  List item link and description must be separated with a dash        remark-lint:awesome/list-item
  ✖  87:3   Invalid list item link                                              remark-lint:awesome/list-item
  ✖  87:5   Link to https://www.nitrous.io is dead                              remark-lint:no-dead-urls
  ✖  88:3   Invalid list item link                                              remark-lint:awesome/list-item
  ✖  89:3   Invalid list item link                                              remark-lint:awesome/list-item

  39 errors
```

https://github.com/sindresorhus/awesome-lint

File: readme.md (hunk 1 above)
Comment (sindresorhus) [reply readme.md]: `Online IDEs` => `Online IDE` to be consistent with the actual list.

**Comment by sindresorhus** [comment]
@styfle Ping :)

**Comment by styfle** [comment]
Hi @sindresorhus thanks for taking another look!

The reason why my readme fails lint is because you have updated `awesome-lint` [7 times](https://www.npmjs.com/package/awesome-lint?activeTab=versions) since I made this PR 😮 

Realistically, it doesn't make sense for me to keep bumping the `awesome-lint` dependency and trying to comply to the changing style guide. I don't want to spend hours maintaining this list AND then hours maintaining the style guide. So I pinned the version of `awesome-lint` and maybe one day I'll update it.


https://github.com/styfle/awesome-online-ide/blob/master/.travis.yml#L9

**Comment by sindresorhus** [comment]
> Realistically, it doesn't make sense for me to keep bumping the awesome-lint dependency and trying to comply to the changing style guide.

It hasn't really changed as much as the linter has just gotten better and stricter.

> I don't want to spend hours maintaining this list AND then hours maintaining the style guide. So I pinned the version of awesome-lint and maybe one day I'll update it.

I totally understand. Time is limited. But Awesome lists are expected to be well maintained and follow the linter for consistency and ease of contribution for users.

I'm gonna close this for now, but happy to review again in the future when you have more time to fix the style issues and maintain it.

**Pull request closed without merging**